test/e2e/apps/job.go

```diff
@@ -344,6 +344,55 @@ var _ = SIGDescribe("Job", func() {
 		}
 	})
 
+	ginkgo.It("should recreate pods only after they have failed if pod replacement policy is set to Failed", func(ctx context.Context) {
+		ginkgo.By("Creating a job")
+		job := e2ejob.NewTestJob("notTerminate", "pod-recreate-failed", v1.RestartPolicyNever, 1, 1, nil, 1)
+		job.Spec.Template.Finalizers = []string{"fake.example.com/blockDeletion"}
+		job.Spec.PodReplacementPolicy = ptr.To(batchv1.Failed)
+		job, err := e2ejob.CreateJob(ctx, f.ClientSet, f.Namespace.Name, job)
+		framework.ExpectNoError(err, "failed to create job in namespace: %s", f.Namespace.Name)
+		defer func() { _ = removePodsFinalizer(ctx, f.ClientSet, f.Namespace.Name) }()
```

Comment:
@alculquicondor what is your opinion on handling errors here? This isn't perfect do disregard the error but on the other side, we cannot do much about it.

Reply:
I think you could use the expectNoError and not ignore it. I think the linter may complain about us not checking the err.